inbm-lib/inbm_common_lib/utility.py

```diff
@@ -205,7 +205,9 @@ def is_within_directory(directory: str, target: str) -> bool:
     return prefix == abs_directory
 
 
-def safe_extract(tarball: tarfile.TarFile, path: str = ".", members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
+def safe_extract(tarball: tarfile.TarFile,
+                 path: str = ".",
+                 members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
     """Avoid path traversal when extracting tarball
```

Comment:
**Suggestion:** To prevent potential directory traversal vulnerabilities, ensure that the 'path' argument is also sanitized before using it in the 'safe_extract' function. [security]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
def safe_extract(tarball: tarfile.TarFile,
                 path: str = ".",
                 members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
    """Avoid path traversal when extracting tarball
    @param tarball: tarball to extract
    @param path: path to extract to, which will be sanitized to prevent directory traversal
    @param members: members to extract
    @param numeric_owner: whether to extract numeric owner
    """
    sanitized_path = sanitize_path(path)
    for member in tarball.getmembers():
        member_path = os.path.join(sanitized_path, member.name)
        if not is_within_directory(sanitized_path, member_path):
            raise IOError("Attempted Path Traversal in Tar File")
    tarball.extractall(sanitized_path, members, numeric_owner=numeric_owner)
```